generators/utilities/src/main/resources/csrc/emulator.cc

```diff
@@ -258,14 +258,19 @@ int main(int argc, char** argv)
     return 1;
   }
 
+  int htif_argc = 1 + argc - optind;
+  char** htif_argv = (char **) malloc((htif_argc) * sizeof (char *));
+  htif_argv[0] = argv[0];
+  for (int i = 1; optind < argc;) htif_argv[i++] = argv[optind++];
```

Comment:
I think you need to do something more clever here to have this still work with permissive args see #563 
I could be wrong but you should test that you can still pass permissive args in.

Reply:
Sure. I copied this code from `emulator.cc` before the multithreading commit, and it should work if the code worked before. I will check it anyway though to make sure.